bibcheck/plugins/journal_names_new.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""
+Checking records in journals collection
+exactly one full name (130__a)
+exactly one short name (711__a)
+all full names, short names, codens (030__a) and name variants (730__a) are unique
+Option sort_variants will sort 730 fields by length.
+"""
+def normalize_name(name):
+    return name.replace('. ', ' ').replace('.', ' ').strip().upper()
+
+def sorted_keys(dd):
+    """return keys of a dict sorted by list of values"""
+    def by_value(a,b):
+        for i in range(len(dd[a])):
+            comp = cmp(dd[a][i], dd[b][i])
+            if comp != 0:
+               return comp
+        return comp
+    ll = dd.keys()
+    ll.sort(cmp=by_value)
+    return ll
+
+from invenio.search_engine import get_collection_reclist, get_fieldvalues, get_record
+from invenio.bibcheck_task import AmendableRecord
+
+JNAMES = {}
+for recid in get_collection_reclist('Journals'):
+    for name in set(get_fieldvalues(recid, '711__a') + get_fieldvalues(recid, '130__a')):
+        norm_name = normalize_name(name)
+        if norm_name in JNAMES:
+            if JNAMES[norm_name] != recid: # short and long name can be identical
+                record = AmendableRecord(get_record(recid))
+                record.set_invalid('Normalized name "%s" (%s) exists in other record %s.' % (norm_name, recid, JNAMES[norm_name]))
```

Comment:
**not sure if this works!**
Maybe there is other initialization needed.
Or it might be overwritten when the record is processed in the normal loop.